Mathlib/Order/RelIso/Set.lean

```diff
@@ -107,6 +107,38 @@ theorem RelIso.preimage_eq_image_symm (e : r ≃r s) (t : Set β) : e ⁻¹' t =
 
 end image
 
+/-- A relation is isomorphic to the lexicographic sum of elements less than `x` and elements not
+less than `x`. -/
+def sumLexLtLE (r : α → α → Prop) [IsTrans α r] [IsTrichotomous α r] [DecidableRel r] (x : α) :
+    r ≃r Sum.Lex (Subrel r {y | r y x}) (Subrel r {y | ¬ r y x}) where
+  toFun y := if h : r y x then Sum.inl ⟨y, h⟩ else Sum.inr ⟨y, h⟩
```

Comment:
Isn't there an existing `Equiv` here for splitting a type by a predicate?

Reply:
I think [`Equiv.Set.sumCompl`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Logic/Equiv/Set.html#Equiv.Set.sumCompl) is the one that you want.

Reply:
Which suggests this should be called some permutation of the tokens `sum`, `compl`, `subrel`, and `lex`.